File: ruby/phone-number/phone_number.rb
# PhoneNumber module comment
module PhoneNumber
  def self.clean(number)
    clear_num = number.scan(/\d/).join
    clear_num = clear_num[1..-1] if clear_num.country_code?

    return nil unless clear_num.valid_format?

    clear_num
  end

  private

  def country_code?
    size == 11 && self[0].to_i == 1
  end

  def valid_format?
    size == 10 && self[0].to_i.between?(2, 9) && self[3].to_i.between?(2, 9)
  end
end

String.class_eval { include PhoneNumber }

File: python/phone-number/phone_number.py
import re

def __version__():
  VERSION = 1

class Phone(object):
  def __init__(self, phone_number):
    self._filter_number = ''.join(re.findall("\d", phone_number))
    if self.__hascountrycode__():
      self._filter_number = self._filter_number[1:]

    if self.__validformat__():
      self.number = self._filter_number
      self.area_code = self._filter_number[:3]
    else:
      raise ValueError("invalid number")

  def pretty(self):
    return f"({self.area_code}) {self.number[3:6]}-{self.number[6:]}"
    
  def __hascountrycode__(self):
    return len(self._filter_number) == 11 and int(self._filter_number[0]) == 1

  def __validformat__(self):
    return len(self._filter_number) == 10 and int(self._filter_number[0]) in range(2, 9) and int(self._filter_number[3]) in range(2, 9)
